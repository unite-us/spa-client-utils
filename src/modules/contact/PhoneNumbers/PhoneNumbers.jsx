import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from '@unite-us/ui';
import phones from 'modules/utils/phones';

const PhoneNumbers = ({ phone_numbers = [], hideLabels, isInNetworkGroup, sortOrder, detailedExtension }) => {
  const phoneNumbers = phones.sortPhoneNumbers(phone_numbers, sortOrder);
  return (
    phoneNumbers.map((phoneNumber) => {
      const {
        extension = '',
        phone_type = '',
        phone_number = '',
      } = phoneNumber;

      const capitalized_phone_type = phone_type ? phone_type[0].toUpperCase() + phone_type.slice(1) : '';
      let numberText = (!hideLabels && capitalized_phone_type) ? `${capitalized_phone_type}: ` : '';
      numberText += phones.formatPhoneNumber(extension ? phone_number + extension : phone_number, detailedExtension);
      return (
        <div
          key={`phone-${phone_number}`}
          className="ui-contact-information__phone mb-half"
        >
          <a
            href={phones.getTelLink(phoneNumber, detailedExtension)}
            target="_blank"
            tabIndex={0}
            rel="noopener noreferrer"
          >
            <Icon
              className="program-details__icon program-details__icon--phone mr-quarter"
              color={`${isInNetworkGroup ? '#4571BA' : '#2C405A'}`}
              icon="V2Phone"
            />

            <span
              className="ui-provider-card__icon-text--phone ml-half normal-case"
              {...(detailedExtension && { 'data-test-element': 'client_phone_number' })}
            >
              {numberText}
            </span>
          </a>
        </div>
      );
    }));
};

PhoneNumbers.propTypes = {
  hideLabels: PropTypes.bool,
  isInNetworkGroup: PropTypes.bool,
  phone_numbers: PropTypes.array.isRequired,
  sortOrder: PropTypes.array,
  detailedExtension: PropTypes.bool,
};

PhoneNumbers.defaultProps = {
  hideLabels: false,
  isInNetworkGroup: true,
  sortOrder: ['work', 'mobile', 'home', 'phone', 'fax', 'other', 'unknown'],
  detailedExtension: false,
};

export default PhoneNumbers;
