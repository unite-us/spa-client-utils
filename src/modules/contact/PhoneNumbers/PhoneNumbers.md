```jsx
const phone_numbers = [
  {
    phone_number: '1232221233',
    extension: '1',
    phone_type: 'phone',
  },
  {
    phone_number: '2223123454',
    extension: '1',
    phone_type: 'other',
  },
];

<PhoneNumbers phone_numbers={phone_numbers} />
```
