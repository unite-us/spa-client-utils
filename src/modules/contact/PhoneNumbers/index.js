import PhoneNumbers from './PhoneNumbers';

export default PhoneNumbers;
