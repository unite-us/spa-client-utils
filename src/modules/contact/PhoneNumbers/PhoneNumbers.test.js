import React from 'react';
import renderer from 'react-test-renderer';
import PhoneNumbers from './PhoneNumbers';

describe('PhoneNumbers', () => {
  const props = {
    phone_numbers: [
      {
        phone_number: '1232221233',
        extension: '1',
        phone_type: 'phone',
      },
      {
        phone_number: '2223123454',
        extension: '1',
        phone_type: 'other',
      },
    ],
  };

  it('renders Program Eligibility', () => {
    expect(renderer.create(<PhoneNumbers {...props} />)).toMatchSnapshot();
  });

  it('renders phone_type property', () => {
    const instance = renderer.create(<PhoneNumbers {...props} />);
    const firstPhoneNumberText = instance.root
      .findAllByProps({ className: 'ui-provider-card__icon-text--phone ml-half normal-case' })[0]
      .children;
    expect(firstPhoneNumberText).toEqual(['Phone: (123) 222-1233 1']);
  });

  it('does not render phone_type property when hideLabels prop is true', () => {
    const instance = renderer.create(<PhoneNumbers {...props} hideLabels />);
    const firstPhoneNumberText = instance.root
      .findAllByProps({ className: 'ui-provider-card__icon-text--phone ml-half normal-case' })[0]
      .children;
    expect(firstPhoneNumberText).toEqual(['(123) 222-1233 1']);
  });
});
