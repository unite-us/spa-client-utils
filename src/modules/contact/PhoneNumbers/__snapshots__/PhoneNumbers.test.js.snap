// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneNumbers renders Program Eligibility 1`] = `
Array [
  <div
    className="ui-contact-information__phone mb-half"
  >
    <a
      href="#"
      rel="noopener noreferrer"
      tabIndex={0}
      target="_blank"
    >
      <svg
        className="ui-icon program-details__icon program-details__icon--phone mr-quarter"
        height="16px"
        id=""
        style={
          Object {
            "display": "inline-block",
            "fill": "#4571BA",
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 13 13"
        width="16px"
      >
        <path
          d="M10.9375 0.625H1.3125C0.574219 0.625 0 1.22656 0 1.9375V11.5625C0 12.3008 0.574219 12.875 1.3125 12.875H10.9375C11.6484 12.875 12.25 12.3008 12.25 11.5625V1.9375C12.25 1.22656 11.6484 0.625 10.9375 0.625ZM2.57031 11.125C2.35156 11.125 2.1875 11.0156 2.16016 10.8242L1.75 9.04688C1.75 9.01953 1.75 8.96484 1.75 8.9375C1.75 8.80078 1.85938 8.63672 1.99609 8.58203L3.91016 7.76172C3.9375 7.73438 4.01953 7.73438 4.04688 7.73438C4.15625 7.73438 4.32031 7.78906 4.375 7.87109L5.22266 8.91016C6.5625 8.28125 7.62891 7.1875 8.25781 5.875L7.21875 5.02734C7.13672 4.97266 7.08203 4.80859 7.08203 4.72656C7.08203 4.67188 7.08203 4.58984 7.10938 4.5625L7.92969 2.64844C7.98438 2.51172 8.14844 2.40234 8.28516 2.40234C8.3125 2.40234 8.36719 2.40234 8.39453 2.40234L10.1719 2.8125C10.3359 2.83984 10.5 3.03125 10.5 3.19531C10.5 7.59766 6.94531 11.125 2.57031 11.125Z"
        />
      </svg>
      <span
        className="ui-provider-card__icon-text--phone ml-half normal-case"
      >
        Phone: (123) 222-1233 1
      </span>
    </a>
  </div>,
  <div
    className="ui-contact-information__phone mb-half"
  >
    <a
      href="#"
      rel="noopener noreferrer"
      tabIndex={0}
      target="_blank"
    >
      <svg
        className="ui-icon program-details__icon program-details__icon--phone mr-quarter"
        height="16px"
        id=""
        style={
          Object {
            "display": "inline-block",
            "fill": "#4571BA",
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 13 13"
        width="16px"
      >
        <path
          d="M10.9375 0.625H1.3125C0.574219 0.625 0 1.22656 0 1.9375V11.5625C0 12.3008 0.574219 12.875 1.3125 12.875H10.9375C11.6484 12.875 12.25 12.3008 12.25 11.5625V1.9375C12.25 1.22656 11.6484 0.625 10.9375 0.625ZM2.57031 11.125C2.35156 11.125 2.1875 11.0156 2.16016 10.8242L1.75 9.04688C1.75 9.01953 1.75 8.96484 1.75 8.9375C1.75 8.80078 1.85938 8.63672 1.99609 8.58203L3.91016 7.76172C3.9375 7.73438 4.01953 7.73438 4.04688 7.73438C4.15625 7.73438 4.32031 7.78906 4.375 7.87109L5.22266 8.91016C6.5625 8.28125 7.62891 7.1875 8.25781 5.875L7.21875 5.02734C7.13672 4.97266 7.08203 4.80859 7.08203 4.72656C7.08203 4.67188 7.08203 4.58984 7.10938 4.5625L7.92969 2.64844C7.98438 2.51172 8.14844 2.40234 8.28516 2.40234C8.3125 2.40234 8.36719 2.40234 8.39453 2.40234L10.1719 2.8125C10.3359 2.83984 10.5 3.03125 10.5 3.19531C10.5 7.59766 6.94531 11.125 2.57031 11.125Z"
        />
      </svg>
      <span
        className="ui-provider-card__icon-text--phone ml-half normal-case"
      >
        Other: (222) 312-3454 1
      </span>
    </a>
  </div>,
]
`;
